python/datashaper/datashaper/engine/verbs/verbs_mapping.py

```diff
@@ -31,6 +43,98 @@ def inner(func: Callable[..., TableContainer]) -> Callable[..., TableContainer]:
     return inner
 
 
+def new_row(old_tuple: tuple, new_column_name: str, value: Any) -> tuple:
+    """Helper function to return rows in row-wise operations."""
+    old_named_tuple_type = type(old_tuple)
+    return namedtuple(
+        old_named_tuple_type.__name__, old_named_tuple_type._fields + (new_column_name,)
+    )(*(old_tuple + (value,)))
+
+
+def parallel_verb(
+    name: str,
+    treats_input_tables_as_immutable: bool = False,
+    override_existing: bool = False,
+    asyncio_type: str = "asyncio",
+    **kwargs,
+) -> Callable:
+    """Decorator for registering a parallel verb."""
+
+    def inner(func: Callable[..., pd.DataFrame]) -> Callable[..., TableContainer]:
+        @verb(
+            name=name,
+            treats_input_tables_as_immutable=treats_input_tables_as_immutable,
+            override_existing=override_existing,
+        )
+        async def wrapper(
+            input: VerbInput,
+            callbacks: Any,
+            max_parallelism: int = 4,
+            *args,
+            **kwargs,
+        ):
+            input_table = input.source.table
+            chunk_size = kwargs.pop("chunk_size", 1)
+            if chunk_size == 1:
+                chunks = input_table.itertuples()
+            elif (
+                chunk_size > 0
+                and signature(func).parameters["chunk"].annotation == pd.DataFrame
+                and signature(func).return_annotation == pd.DataFrame
+            ):
+                chunks = np.array_split(
+                    input_table, math.ceil(len(input_table) / chunk_size)
+                )
+            else:
+                raise NotImplementedError(
+                    "chunk_size > 1 is only supported for DataFrame inputs and outputs. Use chunk_size = 1 for item-wise operations."
+                )
+
+            tick = progress_ticker(
+                callbacks.progress,
+                num_total=len(chunks) if chunk_size > 1 else len(input_table),
+            )
+            errors = []
+
+            async def execute(
+                chunk: pd.DataFrame | tuple, tick: ProgressTicker, *args, **kwargs
+            ) -> pd.DataFrame | tuple:
+                try:
+                    output = await func(chunk=chunk, *args, **kwargs)
+                    tick(1)
+                    return output
+                except Exception as e:
+                    traceback.print_exc()
+                    errors.append(e)
+                    return None
+
+            futures = [

```

Comment:
Got to be careful here - this method looks like it calls execute() on every chunk by default, and fires off all of these async tasks concurrently. If these tasks hit throttled APIs out-of-the-gate we'll run into over-saturation quickly. We should semaphore the execute call or pass a semaphore into execute

Reply:
I believe this will only create the futures but not execute them to actually execute them you need to call run, gather, or await it. From: https://docs.python.org/3/library/asyncio-task.html#coroutines

> Note that simply calling a coroutine will not schedule it to be executed

Reply:
Oh I was using the semaphore in a wrong way :) fixing now